com.unity.netcode.gameobjects/Editor/Simulator/NetworkScenarioView.cs

```diff
@@ -13,42 +13,87 @@ public class NetworkScenarioView : VisualElement
-                .ToList();
+            Undo.undoRedoPerformed += UndoRedoPerformed;
+            RegisterCallback<DetachFromPanelEvent>(OnDetachedFromPanel);
 
-            var choices = new List<string> { None };
-            choices.AddRange(m_Scenarios.Select(x => x.Name));
+            m_Scenarios = FindScenarios();
+            m_Choices = new() { None };
+            m_Choices.AddRange(m_Scenarios.Select(x => x.Name));
 
-            PresetDropdown.choices = choices;
-            PresetDropdown.index = m_NetworkScenario.NetworkSimulatorScenario == null
-                ? 0
-                : choices.IndexOf(m_NetworkScenario.NetworkSimulatorScenario.GetType().Name);
-            PresetDropdown.RegisterCallback<ChangeEvent<string>>(OnPresetSelected);
+            ScenarioDropdown.choices = m_Choices;
+            ScenarioDropdown.RegisterCallback<ChangeEvent<string>>(OnPresetSelected);
+            UpdateScenarioDropdown();
+        }
+
+        void OnDetachedFromPanel(DetachFromPanelEvent evt)
+        {
+            Undo.undoRedoPerformed -= UndoRedoPerformed;
+        }
+
+        void UndoRedoPerformed()
+        {
+            UpdateScenarioDropdown();
+        }
+
+        void OnScenarioParametersGUIHandler()
+        {
+            m_NetworkScenarioProperty.serializedObject.Update();
+
+            EditorGUI.BeginChangeCheck();
+            EditorGUILayout.PropertyField(m_NetworkScenarioProperty, true);
+            
+            if (EditorGUI.EndChangeCheck())
+            {
+                m_NetworkScenarioProperty.serializedObject.ApplyModifiedProperties();
+            }
         }
 
         void OnPresetSelected(ChangeEvent<string> changeEvent)
         {
             if (changeEvent.newValue == None)
             {
                 m_NetworkScenario.NetworkSimulatorScenario = null;
+                return;
             }
-            else
-            {
-                var scenario = m_Scenarios.First(x => x.Name == changeEvent.newValue);
-                var instance = Activator.CreateInstance(scenario);
-                m_NetworkScenario.NetworkSimulatorScenario = instance as INetworkSimulatorScenario;
-            }
+
+            var scenario = m_Scenarios.First(x => x.Name == changeEvent.newValue);
+            var instance = Activator.CreateInstance(scenario);
+            m_NetworkScenarioProperty.managedReferenceValue = (INetworkSimulatorScenario)instance;
+            m_NetworkScenarioProperty.serializedObject.ApplyModifiedProperties();
+        }
+
+        void UpdateScenarioDropdown()
+        {
+            ScenarioDropdown.index = m_NetworkScenario.NetworkSimulatorScenario == null
+                ? 0
+                : m_Choices.IndexOf(m_NetworkScenario.NetworkSimulatorScenario.GetType().Name);
+        }
+
+        bool TypeIsValidNetworkScenario(Type type)
+        {
+            return type.IsClass && type.IsAbstract == false && typeof(INetworkSimulatorScenario).IsAssignableFrom(type);
```

Comment:
Just curious, is there a reason it has to be `type.IsClass`?

Also, does it also need to be marked `[Serializable]`?

This is nitpicky but I could see some value in breaking this out into a class, like `NetworkScenarioTypeLibrary`